at_client/README.md

```diff
@@ -1,8 +1,103 @@
 
 # at_client
-This SDK provides the essential methods for building an app using the @protocol
\ No newline at end of file
+
+### Introduction
+
+SDK that provides the essential methods for building an app using [The @protocol](https://atsign.com). You may also want to look at [at_client_mobile](https://pub.dev/packages/at_client_mobile).
+
+**at_client** package is written in Dart, supports Flutter and follows the
+@platform's decentralized, edge computing model with the following features: 
+- Cryptographic control of data access through personal data stores
+- No application backend needed
+- End to end encryption where only the data owner has the keys
+- Private and surveillance free connectivity
+
+We call giving people control of access to their data "*flipping the internet*".
+
+## Get Started
+
+Initially to get a basic overview of the SDK, you must read the [atsign docs](https://atsign.dev/docs/overview/).
+
+> To use this package you must be having a basic setup, Follow here to [get started](https://atsign.dev/docs/get-started/setup-your-env/).
+
+Check how to use this package in the [at_client installation tab](https://pub.dev/packages/at_client/install).
+
+## Usage
+
+- Set `AtClientPreferences` to your preferred settings.
+
+```dart
+Directory appSupportDir = await getApplicationSupportDirectory();
+AtClientPreference preferences = AtClientPreference()
+        ..rootDomain = 'root.atsign.org'
+        ..namespace = '.my_namespace'
+        ..hiveStoragePath = appSupportDir.path
+        ..commitLogPath = appSupportDirdir.path
+        ..isLocalStoreRequired = true;
+```
+
+- These `preferences` are used for your application.
+
+```dart
+AtClientManager currentAtSign = await AtClientManager.getInstance().setCurrentAtSign(atSign, AtEnv.appNamespace, preferences);
+```
+
+- Update the user data using the `put()` method.
+
+```dart
+AtKey atKey = AtKey()
+        ..key = 'phone'
+        ..namespace = '.myApp';
+await atClientInstance.put(atKey, '+00 123-456-7890');
+```
+
+- Get the data using the `get()` method.
+
+```dart
+AtKey atKey = AtKey()
+        ..key='phone'
+        ..namespace = '.myApp';
+AtValue value = await atClientInstance.get(atKey);
+print(value.value); // +00 123-456-7890
+```
+
+- Delete the data using the `delete()` method.
+
+```dart
+bool isDeleted = await atClientInstance.delete(atKey);
+print(isDeleted); // true if deleted
+```
+
+- Sync the data to the server using the `sync()` method if needed.
+
+```dart
+late SyncService syncService;
```

Comment:
If syncservice is declared late, when does it get initialised?

Reply:
check that now :)